### DISCUSSION TITLE
Open file, open_with. Efficiency and terminal dependencies.

### DISCUSSION THREAD

**Discussion by D0bhareach** [kamiyaa/joshuto#281] [General]
I found joshuto when searched for something like  NERDTree for Helix. Joshuto does great job displaying dirs and files and I used it simply to find and  `yd`  path, but then I wanted more.. What if I use joshuto to open a file in the app of my choice? Joshuto do open files,but.. 
So far I only have tried to open text-based files. Never try to open image of media files. In the core there are couple of methods that actually do open file: `execute_and_wait` and `fork_execute` both are open applications from inside of joshuto process. Which brings two issues: first: if you open modal editor (vim, hx) and try for example kill joshuto because editor becomes kinda disconnected process terminal start to behave strange and cpu start  to execute some sort of loop. `top` command shows`hx` on top of chart, at the end I was have kill it by hand. When file opened with gedit it's OK to close joshuto, but terminal must stay open because gedit is still running on the background; second: efficiency, since joshuto is still active it does require cpu and memory, but if I opened file in vim for example and start to work, chances are joshuto is going to be in the backround for some time, what for?
So far I only think of:    
1. I have been thinking what if I start application separate form joshuto? In for modal editors start new tab/window and pass command and args to it. Didn't try it yet becase it will couple joshuto with specific terminal. It will work for me as far as I won't change terminal. For example gnome-terminal and alacritty have slightly different commands.
2. Try to transform joshuto to add-on for vim, nvim, hx.
3. How apps are started in OS anyway? May be use OS default application dispather? If I find how it shall work for non-modal editors.
So basically two things that I think about: how to pipe messages to System and how to keep joshuto be flexible.
